rosidl_runtime_c/include/rosidl_runtime_c/primitives_sequence_functions.h

```diff
@@ -26,60 +26,112 @@ extern "C"
 {
 #endif
 
-#define ROSIDL_RUNTIME_C__DECLARE_PRIMITIVE_SEQUENCE_FUNCTIONS(STRUCT_NAME, TYPE_NAME) \
+/**
+ * \def ROSIDL_RUNTIME_C__DECLARE_PRIMITIVE_SEQUENCE_INIT(STRUCT_NAME)
+ *
+ * \brief Allocate the memory for the sequence.
+ *
+ * Calling the function with an already allocated sequence will leak the
+ * previously allocated memory.
+ *
+ * param sequence a pointer to a sequence struct
+ * param size the number of items to allocate in the sequence, both sequence
+ *   fields `size` and `capacity` are set to this parameter
+ * return true if successful, false if the passed sequence pointer is null
```

Comment:
Should this be using Doxygen commands, `\param` and `\return`?

Reply:
It can't be since the docblock is for the macro and therefore can't declare the parameter / return value of the function generated by the macro.